tools/regenerate-docs.ts

```diff
@@ -5,7 +5,12 @@ import * as path from 'path';
 import { TSESLint } from '@typescript-eslint/utils';
 import prettier, { Options } from 'prettier';
 import { prettier as prettierRC } from '../package.json';
-import config from '../src/index';
+import plugin from '../src/index';
+import { getRuleNoticeLines } from './rule-notices';
+
+// Marker so that rule doc header (title/notices) can be automatically updated.
+export const END_RULE_HEADER_MARKER =
+  '<!-- end rule header - generated by `yarn tools:regenerate-docs` -->';
```

Comment:
I'm in favor of omitting the `generated by..` because I think its easy to locate the tool by searching for the header and the command might change (i.e. if we extracted it into a cli package, then what do we have here?)

Reply:
Removed this part of the marker.